Refactor PlaybackSessionModel::externalPlaybackTargetType to use enum class

abigailfox opened the pull request:
#### f5f63453c9cda933bfaea5dcae25523ad471293c
<pre>
Refactor PlaybackSessionModel::externalPlaybackTargetType to use enum class
rdar://63360025
<a href="https://bugs.webkit.org/show_bug.cgi?id=242476">https://bugs.webkit.org/show_bug.cgi?id=242476</a>

Reviewed by Eric Carlson.

* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
(): Deleted.
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
(WebCore::PlaybackSessionModelMediaElement::externalPlaybackTargetType const):
* Source/WebCore/platform/ios/PlaybackSessionInterfaceAVKit.mm:
(WebCore::PlaybackSessionInterfaceAVKit::externalPlaybackChanged):
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(VideoFullscreenInterfaceAVKit::~VideoFullscreenInterfaceAVKit):
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
(VideoFullscreenControllerContext::externalPlaybackTargetType const):
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.mm:
(WebKit::PlaybackSessionManagerProxy::externalPlaybackPropertiesChanged):
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.mm:
(WebKit::PlaybackSessionManager::externalPlaybackChanged):

Canonical link: <a href="https://commits.webkit.org/252247@main">https://commits.webkit.org/252247@main</a>
</pre>

eric-carlson reviewed (approved):
Please add bug url(s) to your commit message

abigailfox committed:
Refactor PlaybackSessionModel::externalPlaybackTargetType to use enum class
rdar://63360025
https://bugs.webkit.org/show_bug.cgi?id=242476

Reviewed by Eric Carlson.

* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
(): Deleted.
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
(WebCore::PlaybackSessionModelMediaElement::externalPlaybackTargetType const):
* Source/WebCore/platform/ios/PlaybackSessionInterfaceAVKit.mm:
(WebCore::PlaybackSessionInterfaceAVKit::externalPlaybackChanged):
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(VideoFullscreenInterfaceAVKit::~VideoFullscreenInterfaceAVKit):
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
(VideoFullscreenControllerContext::externalPlaybackTargetType const):
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.mm:
(WebKit::PlaybackSessionManagerProxy::externalPlaybackPropertiesChanged):
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.mm:
(WebKit::PlaybackSessionManager::externalPlaybackChanged):

Canonical link: https://commits.webkit.org/252247@main
